manifests/register.pp

```diff
@@ -0,0 +1,26 @@
+# @summary Register the host as foreman
+# @api private
+class foreman::register (
+  Stdlib::Fqdn $foreman_host_name = $facts['networking']['fqdn'],
+) {
+  if $foreman::register_in_foreman {
```

Comment:
Feels a little odd to have the `if` here as this basically makes the class pointless for someone that included it. I think it also makes the code slightly harder to read for me. When I compare it to:

```
if $foreman::register_in_foreman {
  include foreman::register
}
```